### DISCUSSION TITLE
Is it possible to apply migrations written in pure SQL?

### DISCUSSION THREAD

**Discussion by frederikhors** [SeaQL/sea-query#437] [Q&A] [answered]
I'm coming from Go. I was using `github.com/golang-migrate/migrate/v4` that allow me to run migrations written in pure SQL.

Is it possible with Sea orm?

I'm using `sea_orm_migration` but so far I understand that it only performs migrations written in Rust, not simple sql files. True?

I have a lot of .sql files to apply (in order) after database creation.

**Comment by frederikhors** [comment]
I'm trying to use `sqlx` directly only to apply these migrations but I'm having issue using the pool with the below code:

```rust
use sea_orm::{Database};

let db_pool = Database::connect("...").await;

sqlx::migrate!("sql").run(db_pool).await.unwrap();
```

with Cargo.toml:

```toml
...
sqlx = { version = "0.6.1", features = ["runtime-tokio-native-tls", "migrate"] }
```

but the error is:

```
the trait bound `sea_orm_migration::sea_orm::DatabaseConnection: sqlx::Acquire<'_>` is not satisfied
the following other types implement trait `sqlx::Acquire<'c>`:
  <&'c mut sqlx::PgConnection as sqlx::Acquire<'c>>
  <&'c mut sqlx::pool::PoolConnection<sqlx::Postgres> as sqlx::Acquire<'c>>
  <&'c mut sqlx::pool::PoolConnection<sqlx_core::any::database::Any> as sqlx::Acquire<'c>>
  <&'c mut sqlx_core::any::connection::AnyConnection as sqlx::Acquire<'c>>
  <&'t mut sqlx::Transaction<'c, sqlx::Postgres> as sqlx::Acquire<'t>>
  <&'t mut sqlx::Transaction<'c, sqlx_core::any::database::Any> as sqlx::Acquire<'t>>
  <&sqlx::Pool<DB> as sqlx::Acquire<'a>> rustcE0277
main.rs(20, 31): required by a bound introduced by this call
migrator.rs(92, 12): required by a bound in `sqlx::migrate::Migrator::run`
```

Can you help me?

**Comment by billy1624** [comment, marked as answer]
Hey @frederikhors, you'll need to create a `sqlx::Connection` for that to work.

You could also define the SeaORM migration in pure SQL like this: https://www.sea-ql.org/SeaORM/docs/migration/writing-migration/#raw-sql

**Reply by frederikhors** [reply]
I thought you could use the pool that is created by Sea here:

![image](https://user-images.githubusercontent.com/41120635/189480449-cacd0745-0f15-4a88-9b43-ca938e5c15bb.png)

**Reply by billy1624** [reply]
I think you can't access it because of the (private) visibility.

But, yeah, we could if we change the API. Please join the discussion at https://github.com/SeaQL/sea-orm/issues/753
